# `numpy.seterrobj`

> 原文：[`numpy.org/doc/1.26/reference/generated/numpy.seterrobj.html`](https://numpy.org/doc/1.26/reference/generated/numpy.seterrobj.html)

```py
numpy.seterrobj(errobj, /)
```

设置定义浮点错误处理的对象。

错误对象包含定义 NumPy 错误处理行为的所有信息。`seterrobj`在设置错误处理行为的其他函数（`seterr`，`seterrcall`）中被内部使用。

参数：

**errobj** 列表

错误对象，一个包含三个元素的列表：[内部 NumPy 缓冲区大小、错误掩码、错误回调函数]。

错误掩码是一个单个整数，它保存了所有四种浮点错误的处理信息。对于每种错误类型的信息都包含在整数的三位中。如果我们以 8 进制打印它，我们可以看到对于“无效”、“下溢”、“上溢”和“除以零”（按顺序）设置了什么处理方式。打印出来的字符串可以用以下方式解释：

+   0: ‘忽略’

+   1: ‘警告’

+   2: ‘抛出’

+   3: ‘调用’

+   4: ‘打印’

+   5: ‘记录’

另请参阅

`geterrobj`，`seterr`，`geterr`，`seterrcall`，`geterrcall`

`getbufsize`，`setbufsize`

注释

有关浮点异常类型和处理选项的完整文档，请参阅`seterr`。

示例

```py
>>> old_errobj = np.geterrobj()  # first get the defaults
>>> old_errobj
[8192, 521, None] 
```

```py
>>> def err_handler(type, flag):
...     print("Floating point error (%s), with flag %s" % (type, flag))
...
>>> new_errobj = [20000, 12, err_handler]
>>> np.seterrobj(new_errobj)
>>> np.base_repr(12, 8)  # int for divide=4 ('print') and over=1 ('warn')
'14'
>>> np.geterr()
{'over': 'warn', 'divide': 'print', 'invalid': 'ignore', 'under': 'ignore'}
>>> np.geterrcall() is err_handler
True 
```
